src/models/carDao.js

```diff
@@ -0,0 +1,25 @@
+const { appDataSource } = require("./appDataSource");
+const { forClause } = require("../utils/queryModules");
+
+const getCarList = async (offset, limit, brand, type) => {
+  const andClause = forClause([brand, type], ["c.brand_id", "c.type"]);
+
+  return await appDataSource.query(
+    `SELECT
+    c.id,
+    c.brand_id,
+    c.name,
+    c.thumbnail,
+    c.size,
+    c.type,
+    b.name AS brandName
+    FROM 
+    cars c
+    JOIN brands b ON c.brand_id = b.id
+    WHERE TRUE
+    ${andClause}
+    LIMIT ${limit} OFFSET ${offset}`
+  );
+};
```

Comment:
```suggestion
const getCarList = async (offset, limit, brand, type) => {
  const andClause = await forClause([brand, type], ["c.brand_id", "c.type"]);

  return appDataSource.query(
    `SELECT
    c.id,
    c.brand_id,
    c.name,
    c.thumbnail,
    c.size,
    c.type,
    b.name AS brandName
    FROM 
    cars c
    JOIN brands b ON c.brand_id = b.id
    WHERE TRUE
    ${andClause}
    LIMIT ${limit} OFFSET ${offset}`
  );
};
```
의도하신 바를 짐작해보자면, 오히려 위와같이 await코드가 걸려있어야 동기적으로 동작함에 있어서 오류를 최소화할 수 있을 것 같습니다.

Reply:
확인했고 수정했습니다 !